cin_validator/rules/cin2022_23/rule_8770Q.py

```diff
@@ -0,0 +1,257 @@
+    df_cid_issues = (
+        df_cid.merge(df_merged, left_on="ROW_ID", right_on="ROW_ID")
+        .groupby("ERROR_ID", group_keys=False)["ROW_ID"]
+        .apply(list)
+        .reset_index()
+    )
+
+    # Ensure that you maintain the ROW_ID, and ERROR_ID column names which are shown above. They are keywords in this project.
+    rule_context.push_type_2(
+        table=ChildIdentifiers, columns=[UPN, UPNunknown], row_df=df_cid_issues
+    )
+
+
+def test_validate():
+    # Create some sample data such that some values pass the validation and some fail.
+    sample_cid = pd.DataFrame(
+        [
+            {
+                "LAchildID": "child1",  # 0 Ignore - under 5 years old
+                "PersonBirthDate": "26/05/2000",
+                "UPN": pd.NA,
+                "UPNunknown": pd.NA,
+            },
+            {
+                "LAchildID": "child2",  # 1 Pass - over 5 with UPN
+                "PersonBirthDate": "26/05/1990",
+                "UPN": "AAA",
+                "UPNunknown": pd.NA,
+            },
+            {
+                "LAchildID": "child3",  # 2 Pass - over 5 with valid UPNunknown
+                "PersonBirthDate": "26/05/1990",
+                "UPN": pd.NA,
+                "UPNunknown": "UN2",
+            },
+            {
+                "LAchildID": "child4",  # 3 Pass - over 5 with valid UPNunknown (lower case)
+                "PersonBirthDate": "26/05/1990",
+                "UPN": pd.NA,
+                "UPNunknown": "un5",
+            },
+            {
+                "LAchildID": "child5",  # 4 Fail - over 5 and no UPN or valid UPNunknown
+                "PersonBirthDate": "26/05/1990",
+                "UPN": pd.NA,
+                "UPNunknown": pd.NA,
+            },
+            {
+                "LAchildID": "child6",  # 5 Fail - over 5 and no UPN or valid UPNunknown (UN1 is not valid for >5yrs)
+                "PersonBirthDate": "26/05/1990",
+                "UPN": pd.NA,
+                "UPNunknown": "UN1",
+            },
+            {
+                "LAchildID": "child7",  # 6 Ignore - over 5 and no UPN or valid UPNunknown (UN8 is not valid) BUT only has ReferralNFA
+                "PersonBirthDate": "26/05/1990",
+                "UPN": pd.NA,
+                "UPNunknown": "UN8",
+            },
+        ]
+    )
+    sample_cin_details = pd.DataFrame(
+        [
+            {"LAchildID": "child1", "ReferralNFA": "0"},
+            {"LAchildID": "child2", "ReferralNFA": "false"},
+            {"LAchildID": "child3", "ReferralNFA": "FALSE"},
+            {"LAchildID": "child4", "ReferralNFA": "0"},
+            {"LAchildID": "child5", "ReferralNFA": "0"},
+            {"LAchildID": "child6", "ReferralNFA": "0"},
+            {"LAchildID": "child6", "ReferralNFA": "1"},
+            {"LAchildID": "child7", "ReferralNFA": "1"},
+        ]
+    )
```

Comment:
So I was wondering, actually, what should happen in a case like child6 has a ReferralNFA of both 0 and 1? Does should 1 value overwrite the importance of the other?